From 54b50f40a77ed6a6de37ae7f8c3fb44c1cd052f6 Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Fri, 5 Nov 2021 15:04:07 -0500
Subject: [PATCH 1/1] ARM:dts:aspeed: Add Ruby dts fan and sb-tsi nodes

emc2305 Fan and sb-tsi CPU0 temp sensor nodes are added with Ruby dts.

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts | 37 +++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index d203d684f94e..cf8364c7623f 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -146,6 +146,16 @@
 	};
 };

+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+};
+
 // Misc Regulators
 &i2c4 {
 	// Net name i2c1
@@ -197,7 +207,34 @@
 		//  port 0 connected to fan controllers which controls 5 fans each
 		i2c@0 {
 			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;

+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
 		};

 		i2c@5 {
--
2.17.1

